### PR TITLE
Improvement[Tool] Add gaode weather query tool

### PR DESCRIPTION
**Pull request by ZhangShenao**
- Add gaode weather query tool.
- Documentation: https://lbs.amap.com/api/webservice/summary
- API keys: https://console.amap.com/dev/index

### CODE REVIEW

File: libs/community/langchain_community/tools/gaode/weather.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    return_direct = True
+
+    def _run(self, *args: Any, **kwargs: Any) -> str:
+        city = kwargs.get("city")
+        if city is None:
+            return "Invalid city input!"
+
+        api_key = os.getenv("GAODE_API_KEY")
+        if api_key is None:
+            return "Please apply for gaode api_key first"
```
Comment (ccurme) [inline libs/community/langchain_community/tools/gaode/weather.py:41]: Should we add an API key to the class attributes and raise an error if the tool is instantiated without one, instead?

Comment (ZhangShenao) [reply libs/community/langchain_community/tools/gaode/weather.py]: OK, I've optimized my implementation.

**Review by baskaryan — COMMENTED** [review]
can we add documentation to docs/docs/integrations/tools/gaode.ipynb?

**Comment by ZhangShenao** [comment]
> can we add documentation to docs/docs/integrations/tools/gaode.ipynb?

OK, I'll add the document later.

**Pull request closed without merging**